arcgis-android-toolkit/src/main/java/com/esri/arcgisruntime/toolkit/floorfilter/FloorFilterManager.kt

```diff
@@ -0,0 +1,349 @@
+        return levels.firstOrNull { isSelectedLevel(it) }
+    }
+
+    /**
+     * Returns the selected [FloorFacility] or null if no [FloorFacility] is selected.
+     *
+     * @since 100.13.0
+     */
+    fun getSelectedFacility(): FloorFacility? {
+        return facilities.firstOrNull { isSelectedFacility(it) }
+    }
+
+    /**
+     * Returns the selected [FloorSite] or null if no [FloorSite] is selected.
+     *
+     * @since 100.13.0
+     */
+    fun getSelectedSite(): FloorSite? {
+        return sites.firstOrNull { isSelectedSite(it) }
+    }
+
+    /**
+     * Loads the [FloorManager] of the [GeoView] attached to the [FloorFilterView]
+     *
+     * @since 100.13.0
+     */
+    fun setupMap(geoView: GeoView, map: GeoModel, setupDone: (() -> Unit)? = null) {
+        this.geoView = geoView
+
+        val floorManager = map.floorManager
+
+        if (floorManager == null) {
+            setupDone?.invoke()
+            return
+        }
+
+        this.floorManager = floorManager
+        val doneLoadingListener: Runnable = object: Runnable {
+            override fun run() {
+                floorManager.removeDoneLoadingListener(this)
+
+                // Do this to make sure the UI gets set correctly if the selected level id was set
+                // before the floor manager loaded.
+                val temp = mSelectedLevelId
+                mSelectedLevelId = null
+                selectedLevelId = temp
+                filterMap()
+
+                setupDone?.invoke()
+            }
+        }
+        floorManager.addDoneLoadingListener(doneLoadingListener)
+        floorManager.loadAsync()
+    }
+
+    /**
+     * Removes the attached [GeoView].
+     *
+     * @since 100.13.0
+     */
+    fun clearGeoView() {
+        clearMapFilter()
+        this.geoView = null
+        this.floorManager = null
+    }
+
+    /**
+     * Zooms to the selected [FloorFacility]. If no [FloorFacility] is selected, it will zoom to
+     * the selected [FloorSite].
+     *
+     * @since 100.13.0
+     */
+    internal fun zoomToSelection() {
+        if (!selectedFacilityId.isNullOrBlank()) {
+            zoomToFacility()
+        } else if (!selectedSiteId.isNullOrBlank()) {
+            zoomToSite()
+        }
+    }
+
+    /**
+     * Zooms to the selected [FloorSite].
+     *
+     * @since 100.13.0
+     */
+    private fun zoomToSite() {
+        zoomToExtent(geoView, getSelectedSite()?.geometry?.extent)
+    }
+
+    /**
+     * Zooms to the selected [FloorFacility].
+     *
+     * @since 100.13.0
+     */
+    private fun zoomToFacility() {
+        zoomToExtent(geoView, getSelectedFacility()?.geometry?.extent)
+    }
+
+    /**
+     * Zooms the [geoView] to the [envelope].
+     *
+     * @since 100.13.0
+     */
+    private fun zoomToExtent(geoView: GeoView?, envelope: Envelope?, bufferFactor:Double = 1.25) {
+        if (geoView != null && envelope != null && !envelope.isEmpty) {
+            try {
```

Comment:
I've had issues in the past with setting the viewpoint on a mapview crashing. I would like to keep this just in case so that if something bad does happen, the app just doesn't zoom to the extent instead of crashing.